Adding -Wstrict-prototypes gate validation (#2251)
Azure/azure-iot-sdk-c: CMakeLists.txt

option(no_logging "disable logging" OFF)
option(use_installed_dependencies "set use_installed_dependencies to ON to use installed packages instead of building dependencies from submodules" OFF)
option(warnings_as_errors  "enable strict compiler warnings-as-errors" ON)

> option(strict_prototypes  "enable GCC strict-prototypes compiler option. This is not supported for our test code." OFF)

Reviewer: ```suggestion
option(strict_prototypes  "enable GCC strict-prototypes compiler option. This is not supported with test code enabled." OFF)
```
